### DISCUSSION TITLE
Can Elsa be used inside commercial application for free?

### DISCUSSION THREAD

**Discussion by tomy2105** [elsa-workflows/elsa-core#798] [Q&A]
I ask because Elsa core license (BSD 3) allows this.

However, I see that you are using some 3rd party components like Tailwind UI (for which you said you have license for Elsa project in https://github.com/elsa-workflows/elsa-core/issues/363). 
Does that license allow others who use this project to use it or we would need to purchase separate license from them?

Are there any other 3rd party components that Elsa use either in runtime or during build time that require separate license if used inside other commercial application?

**Comment by sfmskywalker** [comment]
Very good questions.

Yes, Elsa can be used inside commercial applications for free, including the Tailwind UI components **as it's used by Elsa and its Dashboard**. I asked this question last year, and Adam's response was clear (and pretty awesome) at that: 

> You can MIT/BSD/whatever license your own project and if someone uses your project as a starting point for another project and reuses the components that is unfortunate for us but not the end of the world — we'd rather make the license permissive and not be a pain in the ass for our actual customers than spend our time imagining ways people can abuse it and making the license complicated and difficult to navigate for paying customers.
>
> All we ask is that you only include the components that are actually needed for the project and don't distribute the components unnecessarily 👍

This does mean that if you are planning on extending the Elsa Dashboard with your own screens and would like to use Tailwind UI components that aren't currently provided by the application already, you should purchase a Tailwind UI license obviously.

Checkout their [pricing page](https://tailwindui.com/pricing) for more information - it's pretty permissive:

> **Can I use Tailwind UI for my own commercial projects?**
> Absolutely! Your license gives you permission to build as many of your own projects as you like, whether those are simple public websites or SaaS applications that end users need to pay to access.

Elsa doesn't use any components in its core repo that require separate licenses (if it does, then that would be a mistake that we should fix).
